src/figuresClasses.ts

```diff
@@ -1,19 +1,76 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
 
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    private a: number,
+    private b: number,
+    private c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('This is not a Triangle');
+    }
 
-export class Circle {
+    if (a + b <= c || b + c <= a || c + a <= b) {
+      throw new Error('This is not a Triangle');
```

Comment:
Same error in both cases? Let's try to specify why exactly it cannot be a triangle.
Make your errors more descriptive